### ISSUE TITLE
Exception On Couchbase.Linq.BucketContext.GetDocumentId

### ISSUE THREAD

**Issue by Lightfall** [couchbaselabs/Linq2Couchbase#157]
When KeyAttribute of Class is long GetId method throws exception when called by Save.

public class Product : IEntity
    {
        [Key]
        public long productId { get; set; }
....
}

**Message : Unable to cast object of type 'System.Int64' to type 'System.String'.
StackTrace:    at Couchbase.Linq.BucketContext.GetDocumentId[T](T document)
   at Couchbase.Linq.BucketContext.Save[T](T document)
   at Project.DataManager.SaveAll[T](Expression`1 property, IEnumerable`1 records, Exception& ex)**

Sorry for my bad english.

**Comment by brantburnett** [comment]
@Lightfall 

This is currently a limitation of the key implementation, the property is required to be of type `string`.  As a workaround, I'd recommend adding the following property to your model:

```c#
[Key]
[JsonIgnore]
public string key { get { return productId.ToString(); } } 
```

I will leave this issue open so we can improve this in the future.

Thanks,
Brant

**Comment by brantburnett** [comment]
@Lightfall 

This issue should be resolved now in release 1.2.1.  Thanks for bringing it to our attention.

Brant
